cmd/thanos/compact.go

```diff
@@ -84,6 +84,9 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 	wait := cmd.Flag("wait", "Do not exit after all compactions have been processed and wait for new work.").
 		Short('w').Bool()
 
+	processIndexCache := cmd.Flag("index.process-cache", "Process indexes' cache, upload them to obj store and update metas.").
```

Comment:
We made it compatible with current approach, so I would do it without any flag and always as this is recommended way (no real disadvantage)

Reply:
this is one-time operation, don't think the cache processor has to traverse over index files each time compactor start.

Reply:
Good point, but then we would like to make it explict.. when it is done? When it is safe to turn it off? Can we reword this?

Personally not sure about this now

Reply:
I think I like it to be a flag but we might need more explicit naming. Like `index.generate-missing-cache-files` ? essentially process cache is not very explicit.

And then it's up to a user to run it or not. Might be that user will never run it, so the oldest blocks will not have that index cache and store gw will just do it on its own.

Essentially follow up questions is: 
* How to measure completeness of this process? I guess some `bucket ls` extension that will print things without index-cache? (:

Reply:
>index.generate-missing-cache-files

agree or maybe enable by default and the opposite flag to disable it

>How to measure completeness of this process? I guess some bucket ls extension that will print things without index-cache? (:

rely on metric `thanos_compact_index_cache_missing` ?